src/assignment2/newPost.jsx

```diff
@@ -0,0 +1,65 @@
+import {useState} from "react";
+import { Link } from 'react-router-dom';
+
+const NewPost = () => {
+    const [postName, setPostName] = useState("");
+    const [selectedFile, setSelectedFile] = useState(null);
+    const [msg, setMsg] = useState("");
+
+    /*
+    useEffect(() => {
+       fetch("/posts")
+        .then((response) => response.json())
+        .then((data) => {
+          setPosts(data.posts);
+        })
+        .catch((error) => console.error("Error while fetching data:", error));
+    }, []);
+    */
+
+    //console.log(posts);
+    const handleFileChange = (event) => {
+        setSelectedFile(event.target.files[0]);
+    };
+
+    const handleSubmit = async(event) => {
+
+        event.preventDefault();
+        setMsg("Uploading..!");    
+        const formData = new FormData();
+        formData.append("postName", postName)
+        formData.append("image", selectedFile);
+
+        const resp = await fetch("/posts");
+        const json = await resp.json();
+        const posts1 = json.posts;
+
+        posts1.map(element => {
```

Comment:
nit: incorrect usage of `.map`. Check out https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array
Learn these commonly used methods: `map`, `find`, `filter`, `forEach`